RecoTracker/TkSeedGenerator/src/SeedingOTEDProducer.cc

```diff
@@ -0,0 +1,319 @@
+#include "RecoTracker/TkSeedGenerator/interface/SeedingOTEDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTracker.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTrackerEvent.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/PatternTools/interface/TrajectoryStateUpdator.h"
+
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/MeasurementDet/interface/TrajectoryMeasurementGroup.h"
+
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+SeedingOTEDProducer::SeedingOTEDProducer(edm::ParameterSet const& conf)
+    : theUpdator(nullptr),
+      tkMeasEventToken(consumes<MeasurementTrackerEvent>(conf.getParameter<edm::InputTag>("trackerEvent"))) {
+  vhProducerToken = consumes<VectorHitCollectionNew>(edm::InputTag(conf.getParameter<edm::InputTag>("src")));
+  beamSpotToken = consumes<reco::BeamSpot>(conf.getParameter<edm::InputTag>("beamSpotLabel"));
+  updatorName = conf.getParameter<std::string>("updator");
+  produces<TrajectorySeedCollection>();
+}
+
+SeedingOTEDProducer::~SeedingOTEDProducer() {}
+
+void SeedingOTEDProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("src", edm::InputTag("siPhase2VectorHits", "vectorHitsAccepted"));
+  desc.add<edm::InputTag>("trackerEvent", edm::InputTag("MeasurementTrackerEvent"));
+  desc.add<edm::InputTag>("beamSpotLabel", edm::InputTag("offlineBeamSpot"));
+  desc.add<std::string>("updator", std::string("KFUpdator"));
+  descriptions.add("SeedingOTEDProducer", desc);
+}
+
+void SeedingOTEDProducer::produce(edm::Event& event, const edm::EventSetup& es) {
+  std::unique_ptr<TrajectorySeedCollection> seedsWithVHs(new TrajectorySeedCollection());
+
+  edm::ESHandle<TrackerTopology> tTopoHandle;
+  es.get<TrackerTopologyRcd>().get(tTopoHandle);
+  tkTopo = tTopoHandle.product();
+
+  edm::ESHandle<MeasurementTracker> measurementTrackerHandle;
+  es.get<CkfComponentsRecord>().get(measurementTrackerHandle);
+  measurementTracker = measurementTrackerHandle.product();
+  edm::Handle<MeasurementTrackerEvent> measurementTrackerEvent;
+  event.getByToken(tkMeasEventToken, measurementTrackerEvent);
+
+  layerMeasurements = new LayerMeasurements(*measurementTrackerHandle, *measurementTrackerEvent);
```

Comment:
seems like a memory leak. better to use `unique_ptr`

Reply:
Done